### ISSUE TITLE
Simple Calendar - beta

### ISSUE THREAD

**Issue by eclerkin** [moonstonemedia/Simple-Calendar#2]
I have already posted comments on this plugin here:  https://wordpress.org/support/topic/simple-calendar-30-beta-now-available?replies=5#post-7464218

Examples of what I discovered are attached.
![back_forward_button display - after sc widget dragged into footer widget](https://cloud.githubusercontent.com/assets/13108838/10076661/70956f46-6291-11e5-887c-ea57ceedee8f.png)
![back_forward_button display - before page view](https://cloud.githubusercontent.com/assets/13108838/10076663/70ac49dc-6291-11e5-82f0-abeb65266f61.png)
![back_forward_button display - before sc widget not dragged into footer widget](https://cloud.githubusercontent.com/assets/13108838/10076662/70a8447c-6291-11e5-8ed1-589b6eb846f1.png)

**Comment by nekojira** [comment]
yup, probably it's missing a stylesheet on the last one:

1. does the javascript work (if you click on the buttons)?
2. is the last screenshot from a shortcode, attached calendar, widget or calendar post? 
3. were there other calendars present in the same page?

understood about the `location` tag

thank you

**Comment by eclerkin** [comment]
An explanation of what you are seeing would help.  The first image above is the final rendering after dragging the Simple Calendar widget over to my footer widget - works correctly including navigation buttons.  The second image is the shortcode on a single page. The third image was before I moved Simple Calendar widget over to my Footer widget - navigation buttons were not properly instantiated and not usable.

**Comment by nekojira** [comment]
> The third image was before I moved Simple Calendar widget over to my Footer widge

so, where was it before?

tldr: it's not loading assets (csss/js)

https://github.com/moonstonemedia/Simple-Calendar/blob/master/includes/assets.php#L107

since the calendars could be printed in different positions (own post type, as shortcode on other posts/page, as widget, on homepage...) the plugin needs to tell WordPress to enqueue scripts and styles when one of these conditions is met. Furthermore, the plugin is going to support different calendar types so it's better to avoid load everything, always, on every page or this will inevitably add a lot of bloat.

Most of the logic to decide to enqueue assets is in the method linked just above. It also has filter hooks which extensions, widget can tap into. There's surely room for improvement here. 

Unfortunately, that regex you see in the method has to be there because styles can't be enqueued in a do_shortcode callback, only scripts and shortcode detection has to run before that.

**Comment by eclerkin** [comment]
<blockquote>so, where was it before?</blockquote>
List of "Available Widgets" under dashboard navigation Appearance/Widgets.

**Comment by nekojira** [comment]
Uhm right, but the available widgets are inactive. It's the widgets you can choose from and drag into the sidebar, then they become active and print a calendar in this widget case. So, if the widget was inactive, what was printing that calendar there without css/js?

**Comment by eclerkin** [comment]
I just had the shortcode [calendar id="nnnnn"] pasted into the page without dragging the inactive widget over to a page widget.  I thought the shortcode would work independently of an inactive widget.

**Comment by nekojira** [comment]
yes it should work independently but it was not, hence the bug :)

thanks that's what I wanted to know

**Comment by eclerkin** [comment]
Will there be a feature to modify the stylesheet for events to replace the period in front of the Event name with either nothing, some other graphic and possibly bold or upper case the entire event name?

**Comment by eclerkin** [comment]
I am a little confused on the use of the  Appearance tab/Default Calendar/Span option.  My customer requested that we only show 1 month of Calendar events on the home page.  I selected 1 month for the Span option and then viewed the corresponding calendar.  It shows 1 month on the first view but after clicking the next button it then shows October/November instead of just October. Clicking next button shows multiple months not a single month. 
Here is the Appearance tab:
![appearance_options_set](https://cloud.githubusercontent.com/assets/13108838/10105038/8815378e-6361-11e5-8eca-dd76fe7dd835.png)
Here is the visual of the calendar on the home page (first rendered):
![appearance_options_page_1_view](https://cloud.githubusercontent.com/assets/13108838/10105123/e7de1f00-6361-11e5-91c4-e8e94f446085.png)
Here is the calendar after clicking Next button:
![appearance_options_click_next_multiple_months](https://cloud.githubusercontent.com/assets/13108838/10105147/0a7069a6-6362-11e5-8c3a-7e140268bd85.png)

**Comment by nekojira** [comment]
try setting the calendar start at the beginning of the month. The 'span' starts from where the calendar begins. If it's today (the current day), it might not be the beginning of the month.

ps - it's not recommended you use the beta yet on a production site let alone for customer work

**Comment by eclerkin** [comment]
Ok, will do.  Everything posted here is on my development version only so no worries.  I pulled in our customer yesterday to show them this newer version and they requested a change from the existing production version of GCE.

**Comment by nekojira** [comment]
OK I was just saying that out of your own interest :) 

If you follow some commits I have made here I'm introducing some small changes (and possibly a fix for the problem at the beginning of this ticket). The CSS looks good on your site. I removed the bullets though (they are not very popular in design these days, and they can always be introduced back, even in the event template content).

**Comment by eclerkin** [comment]
No, I didn't know that there are incremental changes to the code.  Should I be downloading these changes to php or css files?  Also, I posted an update on my last comment (see above) regarding Monthly views.

**Comment by eclerkin** [comment]
Just a heads up...PhpED detected a fatal error in SimpleCalendar when I attempted to render the WP instance on my workstation (Windows 7 / XAMPP WP / Apache / PhpED Debug).  I have PHP 5.5.x installed in XAMPP.

Fatal error: Call to undefined function SimpleCalendar\Events\mb_detect_encoding() in C:\Program Files\XAMPP\apps\wordpress\htdocs\wp-content\plugins\simple-calendar\includes\events\event.php on line 226
Here is the code:
<blockquote>
// Event content.</br>
		$title                  = isset( $event['title'] )          ? strip_tags( $event['title'] ) : '';</br>
		$this->title            = esc_attr( iconv( mb_detect_encoding( $title, mb_detect_order(), true ), 'UTF-8', $title ) );</br>

		$description            = isset( $event['description'] )    ? $event['description'] : ''</br>
		$this->description      = esc_attr( iconv( mb_detect_encoding( $description, mb_detect_order(), true ), 'UTF-8', $description ) );
</blockquote>

**Comment by nekojira** [comment]
uhm thanks for the heads up, it seems you don't have that PHP extension installed

that function is there to better support UTF-8 from event sources, but it's something I haven't fully tested (not so much in regards of compatibility but efficacy):
http://php.net/manual/en/function.mb-detect-encoding.php

will think if it's useful, if it is, there's some fallback proposed in PHP.net comments

thanks

**Comment by eclerkin** [comment]
With the new version, the calendar time display is way off.  My recurring event is from 8:00 - 9:00 AM and it is showing 10:53 PM - 10:53PM.  Refreshing the page bumps up the time again to 10:56 PM - 10:56 PM.

**Comment by nekojira** [comment]
you are right there's an error if you use the tags `[start-time]` and `[end-time]` it was working only with `[when]` I'm uploading a fix right now, check Dropbox in a few mins

**Comment by eclerkin** [comment]
What is the URL?

**Comment by eclerkin** [comment]
The time issue has been resolved - thanks.

Also, I have been noticing an inconsistency with the month display in the calendar navigation header.  When the calendar is first rendered, the month display is correct by showing the current month name only.  But once I navigate forward, it will either display the next month value only or show a range of months. I have since cleared the Simple Calendar cache and the Zencache and have not seen it occur yet but it was occurring yesterday.  There is still a problem navigating all the way to the end of calendar events then navigating back and the first month September is not available.  I pasted that screenshot at the bottom.
Here is the first page rendering:
![first_rendering_correct_month_display](https://cloud.githubusercontent.com/assets/13108838/10168364/4c353a98-6681-11e5-849d-76942170b7a6.png)
Here is navigating backward from the last calendar entry to the first and September is not available.
![nav_backward_first_month_no_display](https://cloud.githubusercontent.com/assets/13108838/10168583/4e842768-6682-11e5-827e-13c5d5815a1a.png)

**Comment by nekojira** [comment]
can I have your calendar ID?

**Comment by eclerkin** [comment]
[calendar id="23680"]

**Comment by nekojira** [comment]
no the Google Calendar ID

**Comment by eclerkin** [comment]
Ahh...I was wondering!  ed.clerkin@gmail.com

**Comment by eclerkin** [comment]
I just updated the beta version and I see that you included the actual address from the Google Calendar event instead of the literal [location] - thanks!  It would be great to have a checkbox for location to open Google Maps in a separate page instead of overlaying the existing page.

**Comment by nekojira** [comment]
eventually you should be able to, by passing `[location newwindow="yes"]`

I'm aware that the current tag (shortcodes-like) system is far from optimal, but it was there since the inception of this plugin

eventually in the next iterations we should have theme templates support for theme developers/designers and a WYSIWYG editor that overlays the current system for users

tags are not so intuitive 

in the docs there is a list of available tags and attributes:
http://wpdocs.philderksen.com/google-calendar-events/event-builder/ 

I will update this document later on

**Comment by nekojira** [comment]
if you're following up on wordpress.org forums or subscribed the newsletter, a beta-2 was released yesterday

**Comment by eclerkin** [comment]
I downloaded/installed the Beta 2 but have a question.  I would like to insert a hyperlink to files posted our site whenever a filename is inserted in the Description field (been doing it for years).  I modified google.php to include the hyperlink code but it just comes out as text and not an actual link.  What is the trick to form a valid hyperlink?
![url_insertion](https://cloud.githubusercontent.com/assets/13108838/10497113/10079f44-7279-11e5-9e7c-b7ac7b78b7e6.png)

**Comment by nekojira** [comment]
You should not hack the plugin or you will lose your changes in updates.

You can parse HTML using tags attributes - this is outdated: http://wpdocs.philderksen.com/google-calendar-events/event-builder/event-information-shortcodes/ but it will still use a similar system

[description html="yes"]

will publish updated docs at http://docs.simplecalendar.io

**Comment by eclerkin** [comment]
There is a problem parsing the Google Address in the Location field.  See both screen shots - 1 showing the actual address and the second showing what Google shows as the address.
![bad_google_address](https://cloud.githubusercontent.com/assets/13108838/10498777/5618faf2-7281-11e5-8ce7-5e589a365842.png)
![bad_google_address_2](https://cloud.githubusercontent.com/assets/13108838/10498782/5b9e8b86-7281-11e5-987a-6a3c33c8f242.png)

**Comment by nekojira** [comment]
you can see there's an apostrophe `'` in the location string, those are not valid characters in a URL and are urlencoded.. Then it should be up to Google to understand the URL encoding, evidently he doesn't.  It's possible that sometimes this will fail. Also with ambiguous addresses that Google can't understand. Unfortunately Google Calendar does not save coordinates for events as far as I know and doesn't return them when parsing an events feed. If you have a solution for this feel free to open another issue and post your code suggestions.

**Comment by eclerkin** [comment]
It works fine on our Production site using GCE http://sloworkforce.com but perhaps all of that code has now changed.

**Comment by nekojira** [comment]
I see, then I will look further into it, thanks
